### PR TITLE
APIGOV-18060 - Traceability Agent usage tracking

### PR DESCRIPTION
**Pull request by jcollins-axway**

### CODE REVIEW

File: pkg/config/centralconfig.go [hunk 1]
```diff
@@ -516,6 +571,12 @@ func AddCentralConfigProperties(props properties.Properties, agentType AgentType
 
 	if agentType == TraceabilityAgent {
 		props.AddStringProperty(pathDeployment, "prod", "AMPLIFY Central")
+		props.AddStringProperty(pathLighthouseURL, "https://lighthouse-staging.admin.staging.appctest.com", "URL of the Lighthouse")
+		props.AddStringProperty(pathPlatformEnvironmentID, "", "Platform Environment ID")
+		props.AddBoolProperty(pathPublishUsage, true, "Indicates if the agent can publish usage event to AMPLIFY platform. Default to true")
+		// props.AddBoolProperty(pathPublishMetric, true, "Indicates if the agent can publish metric event to AMPLIFY platform. Default to true")
+		props.AddBoolProperty(pathPublishTraffic, true, "Indicates if the agent can publish traffic event to AMPLIFY platform. Default to true")
+		props.AddDurationProperty(pathEventAggregationInterval, 1*time.Minute, "The time interval at which usage and metric event will be generated.")
```
Comment (vivekschauhan) [inline pkg/config/centralconfig.go:579]: May be we should increase this default time to 5 or 10 mins?

File: pkg/config/centralconfig.go [hunk 2]
```diff
@@ -157,6 +169,7 @@ func NewCentralConfig(agentType AgentType) CentralConfig {
 		SubscriptionConfiguration: NewSubscriptionConfig(),
 		AppendDataPlaneToTitle:    true,
 		UpdateFromAPIServer:       false,
+		EventAggregationInterval:  1 * time.Minute,
```
Comment (vivekschauhan) [inline pkg/config/centralconfig.go:172]: May be we should increase this default time to 5 or 10 mins?

File: pkg/traceability/traceability_test.go [hunk 3]
```diff
@@ -46,6 +46,8 @@ func createCentralCfg(url, env string) *config.CentralConfiguration {
 	authCfg.ClientID = "DOSA_1111"
 	authCfg.PrivateKey = "../transaction/testdata/private_key.pem"
 	authCfg.PublicKey = "../transaction/testdata/public_key"
+	cfg.PublishTrafficEvents = true
+	cfg.EventAggregationInterval = 30 * time.Second
```
Comment (vivekschauhan) [inline pkg/traceability/traceability_test.go:50]: Same here increase default to 5 or 10 mins

**Review by sbolosan — APPROVED** [review]

**Review by sbolosan — APPROVED** [review]

File: pkg/config/centralconfig.go (hunk 1 above)
Comment (jcollins-axway) [reply pkg/config/centralconfig.go]: fixed

File: pkg/config/centralconfig.go (hunk 2 above)
Comment (jcollins-axway) [reply pkg/config/centralconfig.go]: fixed

File: pkg/traceability/traceability_test.go (hunk 3 above)
Comment (jcollins-axway) [reply pkg/traceability/traceability_test.go:50]: this is in a test

**Review by sbolosan — APPROVED** [review]

**Review by vivekschauhan — APPROVED** [review]

**Pull request merged**